apps/maestro/src/server/routers/interchainToken/recordInterchainTokenDeployment.ts

```diff
@@ -43,7 +43,10 @@ export const recordInterchainTokenDeployment = protectedProcedure
       .implementationType()
       .catch(() => null);
 
-    const tokenManagerType = Maybe.of(tokenManagerTypeCode).mapOrNull(
+    const tokenManagerType = Maybe.of(tokenManagerTypeCode).mapOr(
+      // default to mint_burn for interchain tokens
+      // and lock_unlock for canonical tokens
+      input.kind === "canonical" ? "lock_unlock" : "mint_burn",
```

Comment:
can this programmatically cast to one of these enum types instead of hard-coding in this manner? https://github.com/axelarnetwork/axelarjs/blob/main/apps/maestro/src/lib/drizzle/schema/common.ts#L47

Reply:
the type is being validated against the enum union type

Reply:
hmm, so if i follow, can't we just do something like 
`const tokenManagerType = TOKEN_MANAGER_TYPES[tokenManagerTypeCode`?

Reply:
that would work too, but then we'd be hardcoding the default enum index, which is less obvious when reading the code